spacy_streamlit/visualizer.py

```diff
@@ -171,6 +171,43 @@ def visualize_ner(
     colors: Dict[str, str] = {},
     key: Optional[str] = None,
     manual: Optional[bool] = False,
+    displacy_options: Optional[Dict] = {},
+):
+    """
+    Visualizer for named entities.
+
+    doc (Doc, List): The document to visualize.
+    labels (list): The entity labels to visualize.
+    attrs (list):  The attributes on the entity Span to be labeled. Attributes are displayed only when the show_table
+    argument is True.
+    title (str): The title displayed at the top of the NER visualization.
+    colors (Dict): Dictionary of colors for the entity spans to visualize, with keys as labels and corresponding colors
+    as the values. This argument will be deprecated soon. In future the colors arg need to be passed in the displacy_options arg
+    with the key "colors".
+    key (str): Key used for the streamlit component for selecting labels.
+    manual (bool): Flag signifying whether the doc argument is a Doc object or a List of Dicts containing entity span
+    information.
+    displacy_options: Dictionary of options to be passed to the displacy render method for generating the HTML to be rendered.
+    """
+    if not displacy_options:
+        displacy_options = dict()
+    if colors:
+        displacy_options["colors"] = colors
+
+    _visualize_ner(doc, labels=labels, attrs=attrs, show_table=show_table, title=title, key=key, manual=manual,
+                   displacy_options=displacy_options)
+
+
+def _visualize_ner(
```

Comment:
Do we actually need this separate helper here? Can't we just handle this in one single function?

Reply:
This spun off because of discussion here - https://github.com/explosion/spacy-streamlit/pull/31#discussion_r773067077

Reply:
Ahh sorry, that wasn't super clear from the code! Let's do the cleanup and refactor together and in a separate PR later so we don't end up in this in-between state that's potentially confusing.

Reply:
I am not sure I follow. Do we want to go ahead with option 1 from the linked discussion thread?

Reply:
Ah sorry, I didn't mean to confuse things. I was just thinking if we're not raising any warnings yet, we might as well keep it all in one function, not introduce the `_visualize_ner` function yet and just accept that we end up with both `displacy_options` and `color`. Otherwise we end up with a lot of duplication and it's unclear from the code why the `visualize_ner` vs. `_visualize_ner` distinction exists. Does this make sense? So basically, we'd start with something closer to your first suggestion in a patch release and then slowly deprecate the separate arguments later, in favour of `displacy_options`. 

Thanks so much for your work on this btw, this is definitely a nice addition to the API 🙏